lib/db/models/SwapDeal.ts

```diff
@@ -0,0 +1,33 @@
+import Sequelize from 'sequelize';
+import { db } from '../../types';
+
+export default (sequelize: Sequelize.Sequelize, DataTypes: Sequelize.DataTypes) => {
+  const attributes: db.SequelizeAttributes<db.SwapDealAttributes> = {
```

Comment:
The `allowNull` values should all match whether each of these fields is nullable on the `SwapDeal` type, there are a few that don't allow null that can be null on the deal itself currently. (stateReason/errorReason should be nullable but it's not currently, I'll open a PR for that).